shared/naturalcrit/codeEditor/codeEditor.jsx

```diff
@@ -99,6 +101,15 @@ const CodeEditor = createClass({
 		}
 	},
 
+	makeLink : function() {
+		const selection = this.codeMirror.getSelection(), t = selection.slice(0, 1) === '[' && selection.slice(-6) === '](url)';
```

Comment:
`t` (whether to "create" or "undo" the hotkey) is only true if the selection literally ends with `](url)`. This means the "undo" hotkey won't work if the user has changed the url to anything else. i.e., selecting `[](url2)` and pressing the hotkey just creates another link since `url2` != `url`.

Can you change this to accept undoing *any* value instead of requiring exactly `url`?

Reply:
So this was something I fiddled with a bit, but was left wondering if the end result of undoing should be:
```
a) [alt text](duckduckgo.com)  -->  alt text duckduckgo.com
b) [alt text](duckduckgo.com)  -->  alt text
c) [alt text](duckduckgo.com)  -->  duckduckgo.com
d) something else?
```
I was going to post about it and ask for input, but then I checked to see what Github does in the same situation...and it actually has the same behavior as this PR currently has.  
```
[alt text](duckduckgo.com) -->  [[alt text](duckduckgo.com)](url)
```
So I thought if it's good enough for GitHub, it's good enough for and left it at that.  But it can be updated to another option, I'm just interested to hear which is preferred.  In my mind, no information should be lost, so option (a) would be my choice (if any change happens at all).

Reply:
I would prefer option A if you can do it.

The issue I was trying to address though, was that `[alt text](duckduckgo.com)` turns into `[[alt text](duckduckgo.com)](url)`, but `[alt text](url)` turns into `url`. The behavior should be the same whether the url is "url" or "duckduckgo".

Reply:
Okay check it out now.  It uses regex to sort out anything in the URL field, and I went for option A above as the desired result.  It also finishes the "undoing" of a hyperlink with the URL selected, so the next step of deleting that URL is a little easier.

```
alt text  -->  [alt text](url)
[alt text](duckduckgo.com)  -->  alt text duckduckgo.com
*blank*  -->  [](url)
[](url)  -->  url
```

Some more changes I was fiddling with but couldn't figure out:

A) regex to determine if the selection is inbetween `](` and `)` so that link syntax could be removed by just having the URL highlighted and doing the hotkey.
B) just cleaning the code up a bit and using the capture groups to assign the `altText` and `url` constants.
C) should a blank zero-length selection end up creating `[alt text](url)`....adding "alt text" in as a placeholder rather than just empty brackets?

Reply:
I added parts B and C that you suggested. Part A would be nice but is a little more involved because we need to grab text *around* the selection which I'd need to look into. If people complain we can change that.